src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -348,6 +348,17 @@ public bool IsEquivalentTo(SyntaxNode other)
             return this.Green.IsEquivalentTo(other.Green);
         }
 
+        /// <summary>
+        /// Returns true if these two nodes are considered "incrementally identical".  An incrementally identical node
+        /// occurs when a <see cref="SyntaxTree"/> is incrementally parsed using <see cref="SyntaxTree.WithChangedText"/>
+        /// and the incremental parser is able to take the node from the original tree and use it in its entirety in the
+        /// new tree.  In this case, the <see cref="SyntaxNode.ToFullString()"/> of each node will be the same, though 
+        /// will have different parents, and may occur at different positions in the respective trees.  If two nodes are
+        /// incrementally identical, all children of each node will be incrementally identical as well.
+        /// </summary>
+        public bool IsIncrementallyIdenticalTo(SyntaxNode other)
+            => this.Green == other.Green;
```

Comment:
Do we need a null check for `other`, like we have for `IsEquivalentTo`?

Reply:
And do we also want to use this API in things like `IsEquivalentTo`?

Reply:
> Do we need a null check for other, like we have for IsEquivalentTo?

I was really on the fence about that.  First, for SyntaxNode, you should never not have a green node.  But, say you didn't... if two nodes don't have a green node, are they incrementally identical!?  I honestly don't know. :D

> And do we also want to use this API in things like IsEquivalentTo?

  Technically, that already happens.  IsEquivalentTo calls GreenNode.IsEquivalentTo, which does an identity check first.

Reply:
> I was really on the fence about that. First, for SyntaxNode, you should never not have a green node. But, say you didn't... if two nodes don't have a green node, are they incrementally identical!? I honestly don't know. :D

Sorry, I meant what if `other` is null? `IsEquivalentTo` will return `false` and not null ref, this will null ref.

Reply:
well played fred.  well played.